src/components/InventoryTable/InventoryTable.js

```diff
@@ -117,128 +79,73 @@ const InventoryTable = forwardRef(
-          onRefreshData({ page: 1 });
-          dispatch(clearErrors());
-        }
-      }
-    }, [error]);
-
-    const cache = useRef(inventoryCache());
-    cache.current.updateProps({
-      page,
-      perPage,
-      items,
-      sortBy,
-      hideFilters,
-      showTags,
-      getEntities,
-      customFilters,
-      hasItems,
-      activeFiltersConfig,
-    });
 
     /**
      * If consumer wants to change data they can call this function via component ref.
      * @param {*} options new options to be applied, like pagination, filters, etc.
      */
-    const onRefreshData = (
-      options = {},
-      disableOnRefresh,
-      forceRefresh = false
-    ) => {
-      const { activeFilters } = store.getState().entities;
-      const cachedProps = cache.current?.getProps() || {};
-      const currPerPage =
-        options?.per_page || options?.perPage || cachedProps.perPage;
+    const onRefreshData = (options = {}, disableOnRefresh) => {
+      const currPerPage = options?.per_page || options?.perPage || perPage;
 
-      const newParams = {
+      const params = {
         page,
         per_page: currPerPage,
-        items: cachedProps.items,
-        sortBy: cachedProps.sortBy,
-        hideFilters: cachedProps.hideFilters,
+        items,
+        sortBy,
+        hideFilters,
         filters: activeFilters,
-        hasItems: cachedProps.hasItems,
-        //RHIF-246: Compliance app depends on activeFiltersConfig to apply its filters.
-        activeFiltersConfig: cachedProps.activeFiltersConfig,
+        hasItems,
+        activeFiltersConfig,
         ...customFilters,
         ...options,
-        globalFilter: cachedProps?.customFilters?.globalFilter,
+        globalFilter: customFilters?.globalFilter,
       };
 
-      //Check for the rbac permissions
-      const cachedParams = cache.current.getParams();
-      if (hasAccess && (!isEqual(cachedParams, newParams) || forceRefresh)) {
-        cache.current.updateParams(newParams);
+      if (hasAccess) {
         if (onRefresh && !disableOnRefresh) {
-          dispatch(entitiesLoading());
-          onRefresh(newParams, (options) => {
+          onRefresh(params, (options) => {
             dispatch(
-              loadSystems(
-                { ...newParams, ...options, controller: controller.current },
-                cachedProps.showTags,
-                cachedProps.getEntities
-              )
+              loadSystems({ ...params, ...options }, showTags, getEntities)
             );
           });
         } else {
-          dispatch(
-            loadSystems(
-              { ...newParams, controller: controller.current },
-              cachedProps.showTags,
-              cachedProps.getEntities
-            )
-          );
+          dispatch(loadSystems(params, showTags, getEntities));
         }
       }
     };
 
-    const prevFilters = useRef(customFilters);
+    const debouncedRefresh = debounce((config) => onRefreshData(config), 800);
```

Comment:
this is the ONE debounced refresh function that should be called. And the only way that any refresh should be triggered.